src/Domain/OrientationMap.cpp

```diff
@@ -13,23 +14,34 @@ OrientationMap<VolumeDim>::OrientationMap() {
     gsl::at(mapped_directions_, j) = Direction<VolumeDim>(j, Side::Upper);
   }
 }
+
 template <size_t VolumeDim>
 OrientationMap<VolumeDim>::OrientationMap(
     std::array<Direction<VolumeDim>, VolumeDim> mapped_directions)
     : mapped_directions_(std::move(mapped_directions)) {
+  std::set<size_t> set_of_dimensions;
```

Comment:
Since `set_of_dimensions` is checked as an `ASSERT` I wonder if would be better to do:
```cpp
#ifdef SPECTRE_DEBUG
  std::set<size_t> set_of_dimensions;
  for (size_t j = 0; j < VolumeDim; j++) {
    set_of_dimensions.insert(gsl::at(mapped_directions_, j).dimension());
  }
  ASSERT(set_of_dimensions.size() == VolumeDim,
         "This OrientationMap fails to map Directions one-to-one.");
#endif
```
Would be better so that `set_of_dimensions` is not present in production code.

Reply:
aren't there 3 sets of these checks?  Couldn't they be factored into a anonymous namespace function that is called in the ASSERTs?

Reply:
Yes, that's cleaner